src/plopp/core/node.py

```diff
@@ -24,18 +49,35 @@ class Node:
     """
 
     def __init__(self, func: Callable, *parents, **kwparents):
-        if not callable(func):
-            raise ValueError("A node can only be created using a callable func.")
-        self.func = func
-        self._id = str(uuid.uuid1())
+        func_is_callable = callable(func)
+        if func_is_callable:
+            self.func = func
+        else:
+            self.func = lambda: func
+        self._id = uuid.uuid4().hex
         self.children = []
         self.views = []
-        self.parents = list(parents)
-        self.kwparents = dict(kwparents)
+        self.parents = [p if isinstance(p, Node) else Node(p) for p in parents]
```

Comment:
In Dask,
```python
d = Delayed(...)
assert d is delayed(d)
```
note the different capitalisation.
If we do this, too, the code here can be simplified to avoid `isinstance` checks.

Reply:
Then you would have something like
```Py
self.parents = [p if p is node(p) else Node(p) for p in parents]
```
?
Not sure it's worth it, using standard Python `isinstance` seems ok to me?

Reply:
No, I mean you can use `node` with a `Node` instance and it returns the same object. So you could write `self.parents = [node(p) for p in parents]`.

Reply:
Would it be the lowercase (decorator) node, or the constructor uppercase `Node`?
`self.parents = [Node(p) for p in parents]` ?

Reply:
The lowercase decorator. Let's not make a constructor return the input object!

Reply:
Now I get it. `node` returns a 'node constructor' not an actual node. Maybe the name isn't very good because
```python
a = node(f)
b = Node(f)

a() # creates a node but does not run any computation
b() # computes a result
```

Reply:
Yeah, but I wanted users to not have to remember too many new names.
I think it kind of makes sense when it's used as a decorator?
```Py
@node
def add(x, y):
    return x + y

my_adding_node = add(input_node(a), input_node(b))
```
The decorator is turning the add function into a node, kind of.

Reply:
> Well something that makes a node anyway.

This is my point.
I'm ok with leaving it as is. But I'm curious if this trips up other people as well.